.github/workflows/sync-from-sunbeam.yaml

```
@@ -0,0 +1,58 @@
+# This workflow is to sync dashboards and alert rules from the Sunbeam repo and create a PR if there is an update.
+name: sync-from-sunbeam
+on:
+  workflow_dispatch:
+  workflow_call:
+  schedule:
+    - cron: '43 1 * * 1'
+
+
+env:
+  REPO_OWNER: canonical
+  REPO_PATH: openstack-exporter-operator
+
+jobs:
+
+  sync-from-sunbeam:
+    name: Pull files from sunbeam
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout local repository
+        uses: actions/checkout@v4
+        with:
+            ref: main
```

I think this should simply be a complete clone.  This feels easier to reason about and should avoid issues with missing refs.

```suggestion
            fetch-depth: 0
```